src/form-builder/components/FormBuilder.jsx

```diff
@@ -85,75 +153,165 @@ export default class FormBuilder extends Component {
     this.props.saveForm(form);
   }
 
-  validateFile(file) {
+  validateFilesInZip(jsonZip) {
     const self = this;
-    const reader = new FileReader();
-    let formData = null;
+    const files = jsonZip.files;
+    const fileNames = (files || files.length < 1) && Object.keys(files);
+    if (fileNames.length < 1) {
+      self.props.onValidationError('Error Importing.. No files found in ZIP');
+      self.hideLoader();
+      return;
+    }
+    const validJsonFileNames = self.getValidJsonFileNames(fileNames);
+    const fileParsePromises = [];
+    const formJsons = [];
+    validJsonFileNames.forEach(fileName => {
+      fileParsePromises.push(jsonZip.file(fileName).async('text').then(data => {
+        try {
+          formJsons.push({ fileName, formData: JSON.parse(data) });
+        } catch (error) {
+          self.updateImportErrors(fileName, self.parseErrorMessage);
+        }
+      }));
+    });
+    Promise.all(fileParsePromises).then(() => self.processForms(formJsons));
+  }
 
+  processForms(formJsons) {
+    const formsValidationPromises = [];
+    const self = this;
+    formJsons.forEach(form => {
+      try {
+        const validateFormJson = self.validateFormJsonAndConcepts(form.fileName, form.formData);
+        if (validateFormJson !== null) {
+          formsValidationPromises.push(validateFormJson);
+        }
+      } catch (e) {
+        this.updateImportErrors(form.fileName,
+          'Parse Error While Importing.. Please import a valid form');
+      }
+    });
+    self.processFormValidationPromises(formsValidationPromises);
+  }
+
+  importJsonFile(file) {
+    const self = this;
+    const reader = new FileReader();
+    const fileName = file[0].name;
     // eslint-disable-next-line
```

Comment:
is the  eslint ignore valid for this line? // eslint-disable-next-line

Reply:
yes

Reply:
Yes